docs/enterprise-api.md

```diff
@@ -835,6 +848,17 @@ message ExecuteWorkflowRequest {
   // overrides will take precedence. Otherwise all env vars set in
   // buildbuddy.yaml will still apply.
   map<string, string> env = 7;
+
+  message ReportingMetadata {
+    string branch = 1;
```

Comment:
For branch name, can we determine the branch name automatically? I think ideally we would be able to make it "just work" if the user passes a commit SHA which exactly corresponds to an existing remote branch tip.

For commit_sha, what's the use case that you have in mind?

Reply:
Here's an example of a situation where we won't be able to determine the branch name automatically:
At the beginning of the CI run, we emit a WorkflowConfigured event with a PushedBranch field, that's used to populate the UI. In order to go from commit sha -> branch name, we need a local clone of the repo (using command `git name-rev --name-only <commit sha>` - I couldn't find a remote command for this). We don't want to delay emitting WorkflowConfigured until after we clone the repo, because that can be slow.

Reply:
Actually I think I may have found a solution here